include/h2o/httpclient.h

```diff
@@ -100,6 +100,7 @@ typedef h2o_httpclient_body_cb (*h2o_httpclient_head_cb)(h2o_httpclient_t *clien
 typedef h2o_httpclient_head_cb (*h2o_httpclient_connect_cb)(h2o_httpclient_t *client, const char *errstr, h2o_iovec_t *method,
                                                             h2o_url_t *url, const h2o_header_t **headers, size_t *num_headers,
                                                             h2o_iovec_t *body, h2o_httpclient_proceed_req_cb *proceed_req_cb,
+                                                            const h2o_header_t **trailers, size_t *num_trailers,
```

Comment:
The way we pass the trailers seem to be diffeent between here and `write_req`. Is it intentional?

Unless there is a reason, it would make sense to use the same way, and based on our convention, that would be passing a list and the size; i.e.,
* `h2o_header_t *` (as each element of the field list is `h2o_header_t`, the passed argument should be a pointer rather than a double pointer), and
* `size_t`denoting the size of the list

Reply:
Addressed in https://github.com/h2o/h2o/pull/3241/commits/f3b3a39aa78384e6de7a12a7e28577b798f77490